spec/API_specification/array_api/array_object.py

```diff
@@ -341,12 +362,18 @@ def __eq__(self: array, other: Union[int, float, bool, array], /) -> array:
 
     def __float__(self: array, /) -> float:
         """
-        Converts a zero-dimensional floating-point array to a Python ``float`` object.
+        Converts a zero-dimensional array to a Python ``float`` object.
+
+        .. note::
+           Casting integer values outside the representable bounds of Python's float type is not specified and is implementation-dependent.
+
+        .. note::
+           When casting a boolean input array, a value of ``True`` must cast to ``1``, and a value of ``False`` must cast to ``0``.
```

Comment:
Same here.

Reply:
Hmm, we don't denote floating values with say a trailing `.` in our special cases throughout the spec (i.e. no "If `x_i` is `0.` or `1.`..."), so feels inconsistent to update this note.

Reply:
I believe this has been resolved. Raising an exception for complex numbers is consistent with `astype`, where the spec does not allow conversion of complex to real.